apps/aecore/src/aec_pow_cuckoo.erl

```diff
@@ -620,15 +623,22 @@ exec_run(Cmd, Dir, Args) ->
     PortName = {spawn_executable, os:find_executable(Cmd, Dir)},
     try
         Port = erlang:open_port(PortName, PortSettings),
-        {os_pid, OsPid} = erlang:port_info(Port, os_pid),
-        ?debug("External mining process started with OS pid ~p", [OsPid]),
-        {ok, Port, OsPid}
+        case erlang:port_info(Port, os_pid) of
+            {os_pid, OsPid} ->
+                ?debug("External mining process started with OS pid ~p", [OsPid]),
+                {ok, Port, OsPid};
+            undefined ->
+                ?warning("External mining process finished before ~p could acquire the OS pid", [?MODULE]),
```

Comment:
I suspect this will mostly happen with the test-miner (16-bit graphs)?!

Reply:
Yes. It actually only happened in system testing. Any real miner would take longer to start.